src/lib.rs

```diff
@@ -0,0 +1,107 @@
+use std::env;
+
+use log::info;
+use tokio::sync::mpsc;
+
+use trin_core::cli::TrinConfig;
+use trin_core::jsonrpc::launch_jsonrpc_server;
+use trin_core::portalnet::protocol::{
+    HistoryNetworkEndpoint, JsonRpcHandler, PortalEndpoint, PortalnetConfig, PortalnetProtocol,
+    StateNetworkEndpoint,
+};
+use trin_history::HistoryRequestHandler;
+use trin_state::StateRequestHandler;
+
+pub async fn main() -> Result<(), Box<dyn std::error::Error>> {
```

Comment:
If this is the top-level orchestrator, would it be better to move this into `main.rs` and remove this `lib.rs` file? Based on the motivation, nothing else would need to call the `main` function of the orchestrator.

Reply:
At the moment, I'd prefer removing `lib.rs` entirely for now under the assumption that if/when the time comes that it's obvious we want to extract some code from `main.rs` then you can always create it later.